### PR TITLE
:seedling: Fixed field alignment

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Fixed field alignment

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: finding/finding.go [hunk 1]
```diff
@@ -96,23 +94,19 @@ const (
...
 type Finding struct {
-	Probe       string             `json:"probe"`
-	Outcome     Outcome            `json:"outcome"`
-	Message     string             `json:"message"`
 	Location    *Location          `json:"location,omitempty"`
 	Remediation *probe.Remediation `json:"remediation,omitempty"`
 	Values      map[string]int     `json:"values,omitempty"`
+	Probe       string             `json:"probe"`
+	Message     string             `json:"message"`
+	Outcome     Outcome            `json:"outcome"`
```
Comment (spencerschrock) [inline finding/finding.go:103]: So one thing about the non-test structs is these values affect the JSON marshaling order. So these `Finding` and `Location` structs may have been intentionally ordered the way they were. @laurentsimon ?

File: rule/rule.go [hunk 2]
```diff
@@ -81,14 +80,13 @@ const (
 	RiskCritical
 )
 
-//nolint:govet
 type Rule struct {
+	Remediation *Remediation
```
Comment (spencerschrock) [inline rule/rule.go:84]: possibly same intentional alignment question

File: finding/finding.go (hunk 1 above)
Comment (laurentsimon) [reply finding/finding.go:103]: I did not order them in particular order at the time. Do we need to?

Comment (spencerschrock) [reply finding/finding.go:103]: It's all preference and readability I think. e.g. if we prefer, a slightly more user-friendly order:
```
    {
      "probe": "toolDependabotInstalled",
      "outcome": 12,
      "message": "tool 'Dependabot' is used",
      "location": {
        "type": 1,
        "path": ".github/dependabot.yml",
        "lineStart": 1
      }
    },
```
or
```
    {
      "location": {
        "type": 1,
        "path": ".github/dependabot.yml",
        "lineStart": 1
      }
      "probe": "toolDependabotInstalled",
      "message": "tool 'Dependabot' is used",
      "outcome": 12,
    },
```

Comment (naveensrinivasan) [reply finding/finding.go:103]: Systems will use these after someone consumes the data.

Comment (spencerschrock) [reply finding/finding.go:103]: I'm fine continuing with this change if we want, since this is an unreleased feature.

**Review by spencerschrock — APPROVED** [review]
We can always revisit in the future if we decide to change the readability approach discussed here. https://github.com/ossf/scorecard/pull/3799#discussion_r1453688928

**Pull request merged**